lib/utils.js

```diff
@@ -3,196 +3,299 @@ var assert = require('assert');
+        f.uses > 0 ? fixts_used.push(i) : null;
+    });
+    assert.deepEqual(fixts_used, expected_used);
+};
 
 // TIMEOUT HELPERS
 
-    timed_out: function(im) {
-        return im.msg.session_event === 'new'
-            && im.user.state.name
-            && im.config.no_timeout_redirects.indexOf(im.user.state.name) === -1;
-    },
+timed_out = function(im) {
+    return im.msg.session_event === 'new'
+        && im.user.state.name
+        && im.config.no_timeout_redirects.indexOf(im.user.state.name) === -1;
+};
+
+timeout_redirect = function(im) {
+    return im.config.timeout_redirects.indexOf(im.user.state.name) !== -1;
+};
+
+// NUMBER HELPERS
 
-    timeout_redirect: function(im) {
-        return im.config.timeout_redirects.indexOf(im.user.state.name) !== -1;
-    },
+// An attempt to solve the insanity of JavaScript numbers
+check_valid_number = function(content) {
+    var numbers_only = new RegExp('^\\d+$');
+    return content !== ''
+        && numbers_only.test(content)
+        && !Number.isNaN(Number(content));
+};
 
+double_digit_number = function(input) {
+    input_numeric = parseInt(input, 10);
+    if (parseInt(input, 10) < 10) {
+        return "0" + input_numeric.toString();
+    } else {
+        return input_numeric.toString();
+    }
+};
 
-// SERVICE API CALL HELPERS
+check_number_in_range = function(input, start, end) {
+    return this.check_valid_number(input) &&
+        (parseInt(input, 10) >= start) &&
+        (parseInt(input, 10) <= end);
+};
 
-    service_api_call: function (service, method, params, payload, endpoint, im) {
+// MSISDN HELPERS
 
-        http = new JsonApi(im, {
-            headers: {
-                'Authorization': ['Token ' + im.config.services[service].api_token]
-            }
-        });
+// Check that it's a number and starts with 0 and approximate length
+// TODO: refactor to take length, explicitly deal with '+'
+is_valid_msisdn = function(content) {
+    return check_valid_number(content)
+        && content[0] === '0'
+        && content.length >= 10
+        && content.length <= 13;
+};
+
+normalize_msisdn = function(raw, country_code) {
+    // don't touch shortcodes
+    if (raw.length <= 5) {
+        return raw;
+    }
+    // remove chars that are not numbers or +
+    raw = raw.replace(/[^0-9+]/g);
+    if (raw.substr(0,2) === '00') {
+        return '+' + raw.substr(2);
+    }
+    if (raw.substr(0,1) === '0') {
+        return '+' + country_code + raw.substr(1);
+    }
+    if (raw.substr(0,1) === '+') {
+        return raw;
+    }
+    if (raw.substr(0, country_code.length) === country_code) {
+        return '+' + raw;
+    }
+    return raw;
+};
+
+// DATE HELPERS
+
+get_timestamp = function(format) {
+    if (format) {
+        return moment().format(format);
+    } else {
+        return moment().format("YYYYMMDDHHmmss");
+    }
+};
+
+// returns today's date if a date not passed in (e.g. format: 'YYYY-MM-DD')
+get_today = function(date, format) {
```

Comment:
This doesn't make sense - the function name is get_today, yet you can pass in a date?